### PR TITLE
Simplify permission requirements for GCP things

### PR DESCRIPTION
**Pull request by sethvargo**
This clears up some confusion around required permissions for the GCP auth method and GCP secrets engines.

### CODE REVIEW

File: website/source/docs/auth/gcp.html.md [hunk 1]
```diff
@@ -10,106 +10,52 @@ description: |-
...
-client before sending an actual request to the Vault server. This process
-**only applies to `iam`-type roles!** For GCE instances, the JWT is
-automatically generated by Google and exposed as metadata.
+Vault includes a CLI helper that obtains a signed JWT locally and sends the
+request to Vault. This helper is only available for IAM-type roles.
 
 ```text
 $ vault login -method=gcp \
     role="my-role" \
     service_account="authenticating-account@my-project.iam.gserviceaccounts.com"
```
Comment (tyrannosaurus-becks) [inline website/source/docs/auth/gcp.html.md]: This isn't changed in this PR, but I do notice it lacks a trailing backslash. Might be nice to add it while we're here.

File: website/source/docs/auth/gcp.html.md [hunk 2]
```diff
@@ -188,51 +134,57 @@ default service account credentials are used.
 For more information on service accounts, please see the [Google Cloud Service
 Accounts documentation][service-accounts].
 
-To use this storage backend, the service account must have the following
-minimum scope(s):
+To use this auth method, the service account must have the following minimum
+scope(s):
```
Comment (tyrannosaurus-becks) [inline website/source/docs/auth/gcp.html.md]:
```suggestion
scope(s):scope
```

**Comment by sethvargo** [comment]
Fixed!

**Comment by sethvargo** [comment]
I don't have merge permissions 😄. Someone else will need to merge and derploy this plz ❤️

**Pull request merged**